src/ng/compile.js

```diff
@@ -1626,6 +1626,37 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     }
 
     /**
+     * A function generator that is used to support both eager and lazy compilation
+     * linking function.
+     * @param eager
+     * @param $compileNodes
+     * @param transcludeFn
+     * @param maxPriority
+     * @param ignoreDirective
+     * @param previousCompileContext
+     * @returns {Function}
+     */
+    function compilationGenerator(eager, $compileNodes, transcludeFn, maxPriority, ignoreDirective, previousCompileContext) {
```

Comment:
If someone makes a change that makes the arguments passed invalid, we should have sufficient unit tests in order to catch any resulting bugs.  Since the compile stage is generally considered a hot path, before making this change I'd be curious to see what ( if any ) perf implications there might be from V8 being unable to optimize this function due to the use of `arguments` like that.

We can also consider copying the arguments manually with a for loop which avoids the deopt if you think it really would make it more resilient.